sdk/tests/conformance/offscreencanvas/offscreencanvas-timer-query.html

```diff
@@ -0,0 +1,84 @@
+<!--
+Copyright (c) 2019 The Khronos Group Inc.
+Use of this source code is governed by an MIT-style license that can be
+found in the LICENSE.txt file.
+-->
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>Test for Timer Query objects with OffscreenCanvas</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+  <div id="description"></div>
+  <div id="console"></div>
+  <script id='myWorker' type='text/worker'>
+  function tick(callback) {
+      function tickImpl() {
+          const res = callback();
+          if (res) {
+              if (requestAnimationFrame) {
+                  requestAnimationFrame(tickImpl);
+              } else {
+                  setTimeout(tickImpl, 10);
+              }
+          }
+      }
+
+      tickImpl();
+  }
+
+  self.onmessage = function(e) {
+      let canvas = new OffscreenCanvas(128, 128);
+      let gl = canvas.getContext("webgl");
+      let ext = gl.getExtension("EXT_disjoint_timer_query");
+      if (!ext) {
+          self.postMessage("PASSED - no EXT_disjoint_timer_query extension - this is legal");
+          return false;
+      }
+      let query = ext.createQueryEXT();
+      ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, query);
+      ext.endQueryEXT(ext.TIME_ELAPSED_EXT);
+      gl.clearColor(0.0, 1.0, 0.0, 1.0);
+      gl.clear(gl.COLOR_BUFFER_BIT);
+      tick(function() {
```

Comment:
FWIW, consider this async pattern:
```
requestAnimationFrame = requestAnimationFrame || function(task) {
  setTimeout(task, 1000/60);
}
async function awaitRaf() {
  await new Promise(res => {
    requestAnimationFrame(() => {
      res();
    });
  });
}
```
```
gl.clear(gl.COLOR_BUFFER_BIT);
while (true) {
  await awaitRaf();
  
  const status = ext.getQueryObjectEXT(query, ext.QUERY_RESULT_AVAILABLE_EXT);
  if (status) {
    self.postMessage("PASSED - timer query object completed successfully on worker");
    break;
  }
  const err = gl.getError();
  if (err != 0) {
    self.postMessage("FAILED - GL error " + err);
    break;
  }
  continue;
}
```

Reply:
Thanks for the suggestion. I'm guessing this would require the onmessage handler to be an async function? Going to leave this as is, but open to a refactoring in the future.